src/Session.js

```diff
@@ -1,162 +1,169 @@
+  return Math.round(
+    ((hrtime[0] + hrtime[1] / 1000 / 1000 / 1000)) * this.rate,
+  ) % 0xffffffff;
 };
 
 Session.prototype.listening = function listening() {
-    this.readyState++;
-    if (this.readyState == 2) {
-        this.emit('ready');
-    }
+  this.readyState += 1;
+  if (this.readyState === 2) {
+    this.emit('ready');
+  }
 };
 
 Session.prototype.handleMessage = function handleMessage(message, rinfo) {
-    log(4, "Incoming Message = ", message);
-    var appleMidiMessage = new ControlMessage().parseBuffer(message),
-        stream;
-    if (appleMidiMessage.isValid) {
-        stream = this.streams.filter(function (stream) {
-            return stream.ssrc == appleMidiMessage.ssrc || stream.token == appleMidiMessage.token;
-        }).pop();
-        this.emit('controlMessage', appleMidiMessage);
-
-
-        if (!stream && appleMidiMessage.command == 'invitation') {
-            stream = new Stream(this);
-            stream.handleControlMessage(appleMidiMessage, rinfo);
-            this.addStream(stream);
-
-        } else if (stream) {
-            stream.handleControlMessage(appleMidiMessage, rinfo);
-        }
-    } else {
-        var rtpMidiMessage = new MidiMessage().parseBuffer(message);
-        stream = this.streams.filter(function (stream) {
-            return stream.ssrc == rtpMidiMessage.ssrc;
-        }).pop();
-        if (stream) {
-            stream.handleMidiMessage(rtpMidiMessage);
-        }
-        this.emit('midi', rtpMidiMessage);
+  logger.verbose('Incoming Message = ', message);
+  const appleMidiMessage = new ControlMessage().parseBuffer(message);
+  let stream;
+  if (appleMidiMessage.isValid) {
+    stream = this.streams.filter(
+      streamItem => (streamItem.ssrc === appleMidiMessage.ssrc)
+        || (streamItem.token === appleMidiMessage.token),
+    ).pop();
+    this.emit('controlMessage', appleMidiMessage);
+
+
+    if (!stream && appleMidiMessage.command === 'invitation') {
+      stream = new Stream(this);
+      stream.handleControlMessage(appleMidiMessage, rinfo);
+      this.addStream(stream);
+    } else if (stream) {
+      stream.handleControlMessage(appleMidiMessage, rinfo);
     }
+  } else {
+    const rtpMidiMessage = new MidiMessage().parseBuffer(message);
+    stream = this.streams.filter(
+      streamItem => streamItem.ssrc === rtpMidiMessage.ssrc,
+    ).pop();
+    if (stream) {
+      stream.handleMidiMessage(rtpMidiMessage);
+    }
+    this.emit('midi', rtpMidiMessage);
+  }
 };
 
 Session.prototype.sendUdpMessage = function sendMessage(rinfo, message, callback) {
-    message.generateBuffer();
-
-    if (message.isValid) {
-        (rinfo.port % 2 == 0 ? this.controlChannel : this.messageChannel).send(message.buffer, 0, message.buffer.length, rinfo.port, rinfo.address, function () {
-          log(4, "Outgoing Message = ", message.buffer, rinfo.port, rinfo.address);
+  message.generateBuffer();
+
+  if (message.isValid) {
+    try {
```

Comment:
Interesting. Do you know why this can fail sometimes?

Reply:
Somehow, when using manager reset, and restarting a session, on _macOS_, there seamed to be socket issues.
````
Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at Socket._healthCheck (dgram.js:638:11)
    at Socket.send (dgram.js:435:8)
    at Session.sendMessage [as sendUdpMessage] (/Users/Nat/Desktop/tmp/rtpmidi/node_modules/rtpmidi/src/Session.js:150:75)
    at Stream.sendReceiverFeedback (/Users/Nat/Desktop/tmp/rtpmidi/node_modules/rtpmidi/src/Stream.js:315:18)
    at ontimeout (timers.js:498:11)
    at tryOnTimeout (timers.js:323:5)
    at Timer.listOnTimeout (timers.js:290:5)
````